fvtest/jitbuildertest/UnionTest.cpp

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include "JBTestUtil.hpp"
+
+union TestUnionInt8pChar
+   {
+   uint8_t v_uint8;
+   const char* v_pchar;
+   };
+
+union TestUnionInt32Int32
+   {
+   uint32_t f1;
+   uint32_t f2;
+   };
+
+union TestUnionInt16Double
+   {
+   uint16_t v_uint16;
+   double v_double;
+   };
+
+DEFINE_TYPES(UnionTypes)
+   {
+   DefineUnion("TestUnionInt8pChar");
+   UnionField("TestUnionInt8pChar", "v_uint8", toIlType<uint8_t>());
+   UnionField("TestUnionInt8pChar", "v_pchar", toIlType<char*>());
+   CloseUnion("TestUnionInt8pChar");
+
+   DefineUnion("TestUnionInt32Int32");
+   UnionField("TestUnionInt32Int32", "f1", Int32);
+   UnionField("TestUnionInt32Int32", "f2", Int32);
+   CloseUnion("TestUnionInt32Int32");
+
+   DefineUnion("TestUnionInt16Double");
+   UnionField("TestUnionInt16Double", "v_uint16", Int16);
+   UnionField("TestUnionInt16Double", "v_double", Double);
+   CloseUnion("TestUnionInt16Double");
+   }
+
+DECL_TEST_BUILDER(TypePunInt32Int32Builder);
+DECL_TEST_BUILDER(TypePunInt16DoubleBuilder);
+
+typedef uint32_t (*TypePunInt32Int32Function)(TestUnionInt32Int32*, uint32_t, uint32_t);
+typedef uint16_t (*TypePunInt16DoubleFunction)(TestUnionInt16Double*, uint16_t, double);
+
+TypePunInt32Int32Builder::TypePunInt32Int32Builder(TR::TypeDictionary *d)
+   : MethodBuilder(d)
+   {
+   DefineLine(LINETOSTR(__LINE__));
+   DefineFile(__FILE__);
+
+   DefineName("typePunInt32Int32Builder");
+   DefineParameter("u", d->PointerTo(d->LookupUnion("TestUnionInt32Int32")));
+   DefineParameter("v1", Int32);
+   DefineParameter("v2", Int32);
+   DefineReturnType(Int32);
+   }
+
+bool
+TypePunInt32Int32Builder::buildIL()
+   {
+   StoreIndirect("TestUnionInt32Int32", "f1", Load("u"), Load("v1"));
+   StoreIndirect("TestUnionInt32Int32", "f2", Load("u"), Load("v2"));
+   Return(
+          LoadIndirect("TestUnionInt32Int32", "f1", Load("u")));
```

Comment:
not a huge deal, but why LoadIndirect on its own line but not Load? seems inconsistent